### ISSUE TITLE
Build CI Pipeline For Release Artifacts

### ISSUE THREAD

**Issue by seancarroll** [doctavious/snippext#6]
Need to build a CI pipeline to build and package release artifacts of CLI for distribution

**Comment by seancarroll** [comment]
Most of the targets initially failed for

> error: could not find system library 'openssl' required by the 'openssl-sys' crate

To see who is pulling in 'openssl-sys' crate

```shell
cargo tree -i openssl-sys
```

Which listed `reqwest` and `git2`. For `reqwest` I was able to avoid pulling `openssl-sys` by adding the following `default-features = false, features = ["blocking", "json", "rustls-tls"]`. For `git2` the only feature I found was `'vendored-openssl'`.

I couldn't find any alternative fix outside of "vendoring" openssl.

Two targets I wasn't able to get working were 
- arm-unknown-linux-musleabihf
- i686-pc-windows-gnu
